### ISSUE TITLE
YOLO9000: better, faster, stronger

### ISSUE THREAD

**Issue by sooooner** [sooooner/papers#44]
# YOLO9000: better, faster, stronger [[link](https://arxiv.org/abs/1612.08242)]

### Abstract
YOLO 성능 개선과 객체 탐지와 분류를 동시에 학습할 수 있는 방법 제안

## Better
YOLO error analysis에서 localization error가 매우 크다는 것을 분석함
Region proposal-based method에 비해 recall이 매우 낮음
즉, recall과 localization을 개선하는데 focus를 맞춤

### Batch Normalization
Regularization을 제거하고 BN을 넣어 성능 향상 (약 2%)
또한 overfitting 걱정없이 Dropout을 제거할수 있게됨

### High Resolution Classifier
ALexNet 이후 대부분의 cls는 224 by 224 에서 pre-train 이후 448에서 detection. 이는 network가 detection으로 동시에 전환하고 새로운 입력 해상도로 조정 해야 함을 의미함

-> 448 by 448 ImageNet에서 10epochs fine-tune 한 다음 detection에 적용 (4%향상)

### Convolutional With Anchor Boxes
Fully Connect을 convolution으로 교체
Anchor box 적용 
mAP 0.3% 떨어졌지만 recall 7% 올라감

### Dimension Clusters
Anchor Box를 추가하면서 box들의 갯수와 비율을 정하기 위해  k-means로 하이퍼 파라미터들을 결정.
k-means에서 거리는 d(box, centroid) = 1 − IOU(box, centroid)를 사용

### Direct location prediction
RCNN계열의 bounding box regression과 비슷하게 예측값들을 다음과 같이 조정함

![image](https://user-images.githubusercontent.com/31437443/113662828-d62dcb00-96e3-11eb-9c09-2976785da165.png)

rcnn계열에서 box regression의 제약을 큰 람다 값으로 주었다면 여기서는 시그모이드를 적용하여 offset 제약을 준것으로 보인다.

![image](https://user-images.githubusercontent.com/31437443/113663134-7ab00d00-96e4-11eb-83da-e876b0e64f77.png)

### Fine-Grained Features
26 x 26 x 512 feature map을 13 x 13 x (512*4)로 변환함(26 x 26을 4개로 나눈뒤 concat)
 
### Multi-Scale Training
iteration이 진행됨에 따라 input size를 다르게 하여 학습함
10번마다 {320, 352, ..., 608}에서 임의로 선택하여 학습

## Faste
기존 feature extractor대신 새로운 feature extract network를 제안함

### Darknet-19

![image](https://user-images.githubusercontent.com/31437443/113665911-4428c100-96e9-11eb-97df-6dd0789a5c29.png)

VGG-16에 비해 크게 차이나지 않지만 연산량을 1/3이하로 줄임

## Stronger
classification과 detection의 joint training
두 task의 data를 통합하는 과정
detection data일때는 기존 loss, classification data 일때는 classification loss만 계산

### Hierarchical classification
두 task의 label의 갭을 줄이기 위해 WordNet기반으로 multi label학습을 함

![image](https://user-images.githubusercontent.com/31437443/113668921-eea2e300-96ed-11eb-90fe-ed54eae80925.png)

## Conclusion
YOLO에 Region proposal-based method의 단점을 보안한 방법을 적용하여 성능과 속도를 향상시킴
